tests/unit/client.tcl

```diff
@@ -0,0 +1,78 @@
+start_server {} {
+    test {CLIENT Caching wrong number of arguments} {
+        catch {r client caching} err
+        set _ $err
+    } {ERR*wrong number of arguments*}
+
+    test {CLIENT Caching wrong argument} {
+        catch {r client caching maybe} err
+        set _ $err
+    } {ERR*when the client is in tracking mode*}
+
+    test {CLIENT Caching OFF without optout} {
+        catch {r client caching off} err
+        set _ $err
+    } {ERR*when the client is in tracking mode*}
+
+    test {CLIENT Caching ON without optin} {
+        catch {r client caching on} err
+        set _ $err
+    } {ERR*when the client is in tracking mode*}
+
+    test {CLIENT Caching ON with optout} {
+        r CLIENT TRACKING ON optout 
+        catch {r client caching on} err
+        set _ $err
+    } {ERR*syntax*}
+    
+    test {CLIENT Caching OFF with optin} {
+        r CLIENT TRACKING off optout 
+        catch {r client caching on} err
+        set _ $err
+    } {ERR*when the client is in tracking mode*}
+
+    test {CLIENT kill wrong address} {
+        catch {r client kill 000.123.321.567:0000} err
+        set _ $err
+    } {ERR*No such*}
+
+    test {CLIENT kill no port} {
+        catch {r client kill 127.0.0.1:} err
+        set _ $err
+    } {ERR*No such*}
```

Comment:
I think the two can be combined. `CLIENT kill wrong address`, `CLIENT kill no port` 
```
if (addr && strcmp(getClientPeerId(client),addr) != 0) continue;
```